bundles/org.eclipse.swt/Eclipse SWT/common/org/eclipse/swt/graphics/ImageGcDrawer.java

```diff
@@ -0,0 +1,50 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Yatta and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Yatta - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.swt.graphics;
+
+/**
+ * Interface to provide a callback mechanism to draw on different GC instances
+ * depending on the zoom the image will be used for. A common use case is when the
+ * application is moved from a low DPI monitor to a high DPI monitor.
+ * This provides API which will be called by SWT during the image rendering.
+ *
+ * This interface needs to be implemented by client code to private logic that draws
+ * on the empty GC on demand.
+ *
+ * @since 3.129
+ */
+public interface ImageGcDrawer {
+
+
+	/**
+	 * Provides a GC to draw on for a requested zoom level.
```

Comment:
This does not describe the method, but what an implementor of the method has to do. The method itself receives a GC and (the implementation) draw on it to provide an image for the requested zoom.
```suggestion
	 * Draws an image on a GC for a requested zoom level.
```

Reply:
done